core/embed/rust/src/ui/model_tt/screens.rs

```diff
@@ -1,33 +1,24 @@
-#[cfg(feature = "micropython")]
-use crate::micropython::buffer::StrBuffer;
-use crate::ui::{
-    component::Component,
-    constant::screen,
-    display,
-    model_tt::{
-        component::{ErrorScreen, WelcomeScreen},
-        constant,
+use crate::{
+    strutil::TString,
+    ui::{
+        component::Component,
+        constant::screen,
+        display,
+        model_tt::{
+            component::{ErrorScreen, WelcomeScreen},
+            constant,
+        },
     },
 };
 
-#[cfg(not(feature = "micropython"))]
-// SAFETY: Actually safe but see below
-unsafe fn get_str(text: &str) -> &str {
-    text
-}
-#[cfg(feature = "micropython")]
-// SAFETY: The caller is responsible for ensuring that the StrBuffer does not
-// escape the lifetime of the original &str.
-unsafe fn get_str(text: &str) -> StrBuffer {
-    unsafe { StrBuffer::from_ptr_and_len(text.as_ptr(), text.len()) }
+fn get_str(text: &str) -> TString {
+    TString::Str(text)
 }
 
 pub fn screen_fatal_error(title: &str, msg: &str, footer: &str) {
-    // SAFETY: these will get placed into `frame` which does not outlive this
-    // function
-    let title = unsafe { get_str(title) };
-    let msg = unsafe { get_str(msg) };
-    let footer = unsafe { get_str(footer) };
+    let title = get_str(title);
+    let msg = get_str(msg);
+    let footer = get_str(footer);
 
     let mut frame = ErrorScreen::new(title, msg, footer);
```

Comment:
remove `get_str` and then
```suggestion
    let mut frame = ErrorScreen::new(title.into(), msg.into(), footer.into());
```

Reply:
d0243af2e2ecd705cf927cfa3220d61d107fc09b